ddp-workspace/projects/ddp-dsm-ui/src/app/sequencing-order/sequencing-order.component.ts

```diff
@@ -0,0 +1,162 @@
+import {Component, Input, ViewChild} from '@angular/core';
+import {ModalComponent} from '../modal/modal.component';
+import {Participant} from '../participant-list/participant-list.model';
+import {Auth} from '../services/auth.service';
+import {ComponentService} from '../services/component.service';
+import {DSMService} from '../services/dsm.service';
+import {RoleService} from '../services/role.service';
+import {PatchUtil} from '../utils/patch.model';
+import {Utils} from '../utils/utils';
+import {SequencingOrder} from './sequencing-order.model';
+
+@Component( {
+  selector: 'app-sequencing-order',
+  templateUrl: './sequencing-order.component.html',
+  styleUrls: [ './sequencing-order.component.scss' ]
+} )
+export class SequencingOrderComponent {
+  @Input() participant: Participant;
+  @Input() samples: SequencingOrder[];
+
+  @ViewChild( ModalComponent )
+  public modal: ModalComponent;
+
+  errorModal = false;
+
+  currentPatchField: string;
+  errorMessage: string;
+  selectedNormal: SequencingOrder = SequencingOrder.getEmptyInstance();
+  selectedTissue: SequencingOrder = SequencingOrder.getEmptyInstance();
+
+  constructor( private dsmService: DSMService, private auth: Auth, private role: RoleService ) {
+
+  }
+
+
+  valueChanged( value: any, parameterName: string, sample: SequencingOrder ): void {
+    let v;
+
+    if (typeof value === 'string') {
+      sample[ parameterName ] = value;
+      v = value;
+    }
+    if (v != null) {
+      const realm: string = localStorage.getItem( ComponentService.MENU_SELECTED_REALM );
+      const patch1 = new PatchUtil(
+        sample.dsmKitRequestId, this.role.userMail(),
+        {
+          name: parameterName,
+          value: v
+        }, null, 'dsmKitRequestId', sample.dsmKitRequestId,
+        'kit', null, realm, null
+      );
+      const patch = patch1.getPatch();
+      this.patch( patch );
+    }
+  }
+
+  patch( patch: any ): void {
+    this.dsmService.patchParticipantRecord( JSON.stringify( patch ) ).subscribe( {
+      next: data => {
+        this.currentPatchField = null;
+      },
+      error: err => {
+        if (err._body === Auth.AUTHENTICATION_ERROR) {
+          this.auth.logout();
+        }
+      }
+    } );
+  }
+
+  isPatchedCurrently( field: string ): boolean {
+    return this.currentPatchField === field;
+  }
+
+  shouldHaveCheckbox( sample: SequencingOrder ): boolean {
+    return ( sample.sampleStatus === 'Received' && sample.sampleType === 'Normal' )
```

Comment:
I would create an enum named SampleType and I would have stored there these string values and call them like this:
SampleType.Received, SampleType.Normal and etc.